components/core/src/clp/BoundedReader.hpp

```diff
@@ -0,0 +1,78 @@
+#ifndef CLP_BOUNDEDREADER_HPP
+#define CLP_BOUNDEDREADER_HPP
+
+#include "ReaderInterface.hpp"
```

Comment:
We are missing some headers includes for:
- ErrorCode
- size_t

Reply:
I know that clang-tidy cares, but isn't the official policy that we don't have to include headers that will definitely be included by the headers that we do include.

If we always defer to clang-tidy then we should change our coding guidelines.

Reply:
- Sorry for the confusion. Can u point me which policy u referred to? Our latest guideline is we should directly include all the headers needed (same as google: https://google.github.io/styleguide/cppguide.html#Include_What_You_Use). I should update this guideline if it's still out-dated.
- I know that clang-tidy is not officially turned on in clp core, but for any newly added files we should try to fix all clang-tidy issues. Kirk and I were trying to proceed clang-tidy in our core, and hopefully we will have it in our automated workflow soon. It makes our life easier if we start following the clang-tidy rules

Reply:
See related section of our style guide https://gitlab.dev.yscope.io/clp/CLP/-/wikis/CPP-style-guide#names-and-order-of-includes

Reply:
I interpret the "unless foo.hpp explicitly demonstrates its intent to provide you the symbols of bar.hpp" clause to cover situations where foo.hpp includes those symbols in the interface that it provides.

Reply:
I think the current style guide is better than strict include what you use on the basis that including headers that are guaranteed to be provided by another header you include wastes vertical space and decreases readability. I would object to changing the style guide in this case.

Reply:
- The guidelines on Gitlab are out of date and we will update them, as well as the notion page
- We want to rely on linters for most things so that we can automate code checks as much as possible. That means we should just stick to one rule that the linter checks
- Adding a few headers recommended by clang-tidy doesn't seem like a huge investment. For now, we have to manually pay the cost of running clang-tidy, but we're working to address that. Also if u're using CLion, clang-tidy should be automatically enabled in the editor to show all warnings

Reply:
I agree in with what you're saying and I will make this change, but please:
1) Keep the style guide up to date
2) Announce changes to the style guide, preferably preceded by RFC

Otherwise what you're asking me to do is strictly follow any style rules you decide on in the moment, which is an absurd coding standard.